xen/drivers/passthrough/arm/ipmmu-vmsa.c

```diff
@@ -1179,6 +1193,16 @@ static void ipmmu_detach_device(struct iommu_domain *io_domain,
 	struct ipmmu_vmsa_archdata *archdata = to_archdata(dev);
 	unsigned int i;
 
+#ifdef CONFIG_HAS_COPROC
+	/*
+	 * Don't allow to even touch uTLBs when deassigning coproc device. Because it
+	 * may be running in other domain at the moment. Only coproc's scheduler
+	 * is allowed to decide when it is time to switch IOMMU context.
+	 */
+	if (dev_iommu(dev)->is_coproc)
```

Comment:
general note: what if there are other users of IOMMU, will we add yet another is_XXX?